tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -320,6 +371,41 @@ export function benchmarkMemory(testObject: IMemoryTestObject): Test {
 						rawValue: runs,
 						formattedValue: prettyNumber(runs, 0),
 					};
+
+					if (baselineMemoryUsage >= 0 && allowedDeviationBytes >= 0) {
+						// Compare the average heap used to the baseline memory usage
+						const avgHeapUsed = heapUsedStats.arithmeticMean;
+						const lowerBound = baselineMemoryUsage - args.allowedDeviationBytes;
+						const upperBound = baselineMemoryUsage + args.allowedDeviationBytes;
+						// Throw errors on regressions/improvements if `ENABLE_MEM_REGRESSION` is set and a warning otherwise.
+						// This allows us to run the tests in CI without failing the build, but still get a warning.
+						if (avgHeapUsed > upperBound) {
+							const message = `Memory Regression detected for test '${
+								testObject.title
+							}': Used '${avgHeapUsed.toPrecision(6)}' bytes, with baseline'${
+								args.baselineMemoryUsage
+							}' and tolerance of '${allowedDeviationBytes}' bytes.\n`;
+							if (ENABLE_MEM_REGRESSION) {
+								process.stderr.write(message);
+								throw new Error(message);
+							} else {
+								process.stdout.write(chalk.yellow(message));
```

Comment:
This one still concerns me. Child processes writing anything to stdout (other than the final test results) feels like it would break the communication between parent and child. Let's double check what happens when things are working (i.e. using `node_modules/` in the reporter path when pnpm-linked) and we hit this line by having a test that goes above/below threshold. I suspect we'll need to remove it, and probably also remove the write to stderr, since the `throw` should already be communicating the problem all the way to the output table.

Reply:
Update: we realized that extra lines in a child process' output don't break things, as long as the child process also outputs one line with the JSON payload that it communicates back to the parent process, so this is not a problem.